app/scripts/metamask-controller.js

```diff
@@ -1195,10 +1195,8 @@ export default class MetamaskController extends EventEmitter {
         this.txController.txGasUtil,
       ),
       networkController: this.networkController,
-      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
-        networkControllerMessenger,
-        NetworkControllerEventType.NetworkDidChange,
-      ),
+      onNetworkStateChange: (listener) =>
```

Comment:
We are about to consolidate these stores, so instead of listening to `networkIdStore`, it might be more future-proof to listen to `networkController.store` and keep track of the current `networkId` and then call the listener if it changes. We can also change this later, so not a big deal, but it would be one less thing to worry about.

Reply:
I'm on my phone at the moment but I'm realizing we may already do this in SwapsController. So maybe we should not do it there? Whatever is easier to understand.

Reply:
Actually... this was already merged, so this probably doesn't work as-written. This may be a strict requirement. Good catch